### ISSUE TITLE
agent capability is not advertised in git-upload-pack when using protocol v2

### ISSUE THREAD

**Issue by syntonyze** [eclipse-jgit/jgit#118]
### Version

6.10.0

### Operating System

MacOS

### Bug description

Perform a `git clone` using a client having protocol v2 enabled against a Gerrit with a git protocol v2 enabled (we used a 
gerrit 3.10.0, where `protocol.version = 2` was configured in `etc/jgit.config`). Use tracing to trace network operations.

```
GIT_TRACE_PACKET=1 git clone "ssh://admin@localhost:29418/test"
```

### Actual behavior

The agent capability is not advertised.

```
GIT_TRACE_PACKET=1 git clone "ssh://admin@localhost:29418/test"
Cloning into 'test'...
15:31:55.205134 pkt-line.c:80           packet:        clone< version 2
15:31:55.207178 pkt-line.c:80           packet:        clone< ls-refs
15:31:55.207186 pkt-line.c:80           packet:        clone< fetch=shallow
15:31:55.207191 pkt-line.c:80           packet:        clone< server-option
15:31:55.207195 pkt-line.c:80           packet:        clone< 0000
15:31:55.207203 pkt-line.c:80           packet:        clone> command=ls-refs
15:31:55.207212 pkt-line.c:80           packet:        clone> 0001
15:31:55.207218 pkt-line.c:80           packet:        clone> peel
15:31:55.207236 pkt-line.c:80           packet:        clone> symrefs
15:31:55.207241 pkt-line.c:80           packet:        clone> ref-prefix HEAD
15:31:55.207244 pkt-line.c:80           packet:        clone> ref-prefix refs/heads/
15:31:55.207248 pkt-line.c:80           packet:        clone> ref-prefix refs/tags/
15:31:55.207250 pkt-line.c:80           packet:        clone> 0000
15:31:55.213012 pkt-line.c:80           packet:        clone< 5dec62ce36a795b4e6738077f69e1fcd9e850138 HEAD symref-target:refs/heads/master
15:31:55.213084 pkt-line.c:80           packet:        clone< 5dec62ce36a795b4e6738077f69e1fcd9e850138 refs/heads/master
15:31:55.213103 pkt-line.c:80           packet:        clone< 0000
15:31:55.217420 pkt-line.c:80           packet:        clone> command=fetch
15:31:55.217439 pkt-line.c:80           packet:        clone> 0001
15:31:55.217443 pkt-line.c:80           packet:        clone> thin-pack
15:31:55.217447 pkt-line.c:80           packet:        clone> ofs-delta
15:31:55.217462 pkt-line.c:80           packet:        clone> want 5dec62ce36a795b4e6738077f69e1fcd9e850138
15:31:55.217467 pkt-line.c:80           packet:        clone> want 5dec62ce36a795b4e6738077f69e1fcd9e850138
15:31:55.217471 pkt-line.c:80           packet:        clone> done
15:31:55.217473 pkt-line.c:80           packet:        clone> 0000
15:31:55.222977 pkt-line.c:80           packet:        clone< packfile
15:31:55.226972 pkt-line.c:80           packet:     sideband< \2Counting objects: 1   \15
15:31:55.227089 pkt-line.c:80           packet:     sideband< \2Counting objects: 2, done
remote: Counting objects: 2, done
15:31:55.227097 pkt-line.c:80           packet:     sideband< \2Finding sources:  50% (1/2)   \15
15:31:55.227104 pkt-line.c:80           packet:     sideband< \2Finding sources: 100% (2/2)   \15
15:31:55.227110 pkt-line.c:80           packet:     sideband< \2Finding sources: 100% (2/2)
remote: Finding sources: 100% (2/2)
15:31:55.227116 pkt-line.c:80           packet:     sideband< PACK ...
15:31:55.227125 pkt-line.c:80           packet:     sideband< \2Total 2 (delta 0), reused 0 (delta 0)
remote: Total 2 (delta 0), reused 0 (delta 0)
15:31:55.227131 pkt-line.c:80           packet:     sideband< 0000
Receiving objects: 100% (2/2), done.
```

### Expected behavior

The server advertises the `agent` capability, as it does when using git protocol v0 or v1.
This is documented [by the protocol](https://git-scm.com/docs/protocol-v2#_agent)

### Other information

This works when setting client or server to work on a git protocol that is not v2, for example, notice how the following advertises the agent:

```
GIT_TRACE_PACKET=1 git clone -c'protocol.version=1' "ssh://admin@localhost:29418/test"
Cloning into 'foofa'...
15:37:13.458082 pkt-line.c:80           packet:        clone< 5dec62ce36a795b4e6738077f69e1fcd9e850138 HEAD\0 include-tag multi_ack_detailed multi_ack ofs-delta side-band side-band-64k thin-pack no-progress shallow agent=JGit/v6.9.0.202403050737-r-23-gc0b415fb0 symref=HEAD:refs/heads/master
15:37:13.460338 pkt-line.c:80           packet:        clone< cc8291fdc783f382d7ed3cb86b917bdab3940bcb refs/changes/01/1/1
15:37:13.460359 pkt-line.c:80           packet:        clone< fd0c9f20131a10cc7cdf68b5347a393a93bf6ddd refs/changes/01/1/meta
15:37:13.460365 pkt-line.c:80           packet:        clone< 5dec62ce36a795b4e6738077f69e1fcd9e850138 refs/heads/master
15:37:13.460370 pkt-line.c:80           packet:        clone< 193a9ce77013f0e9f5b9cf6dfb2f75a920136f7a refs/meta/config
15:37:13.460375 pkt-line.c:80           packet:        clone< 0000
15:37:13.463801 pkt-line.c:80           packet:        clone> want 5dec62ce36a795b4e6738077f69e1fcd9e850138 multi_ack_detailed side-band-64k thin-pack ofs-delta agent=git/2.39.3.(Apple.Git-146)
15:37:13.463827 pkt-line.c:80           packet:        clone> want 5dec62ce36a795b4e6738077f69e1fcd9e850138
15:37:13.463837 pkt-line.c:80           packet:        clone> 0000
15:37:13.463849 pkt-line.c:80           packet:        clone> done
15:37:13.470006 pkt-line.c:80           packet:        clone< NAK
15:37:13.474266 pkt-line.c:80           packet:     sideband< \2Counting objects: 1   \15
15:37:13.474380 pkt-line.c:80           packet:     sideband< \2Counting objects: 2, done
remote: Counting objects: 2, done
15:37:13.474385 pkt-line.c:80           packet:     sideband< \2Finding sources:  50% (1/2)   \15
15:37:13.474389 pkt-line.c:80           packet:     sideband< \2Finding sources: 100% (2/2)   \15
15:37:13.474393 pkt-line.c:80           packet:     sideband< \2Finding sources: 100% (2/2)
remote: Finding sources: 100% (2/2)
15:37:13.474400 pkt-line.c:80           packet:     sideband< PACK ...
15:37:13.474406 pkt-line.c:80           packet:     sideband< \2Total 2 (delta 0), reused 0 (delta 0)
remote: Total 2 (delta 0), reused 0 (delta 0)
15:37:13.474410 pkt-line.c:80           packet:     sideband< 0000
Receiving objects: 100% (2/2), done.
```

Also, performing a git protocol v2 clone against github, shows the agent as expected:

```
GIT_TRACE_PACKET=1 git clone git@github.com:GerritForge/homebrew-gerrit.git
Cloning into 'homebrew-gerrit'...
15:39:56.614665 pkt-line.c:80           packet:        clone< version 2
15:39:56.617324 pkt-line.c:80           packet:        clone< agent=git/github-50ee4bdaf298
15:39:56.617335 pkt-line.c:80           packet:        clone< ls-refs=unborn
15:39:56.617340 pkt-line.c:80           packet:        clone< fetch=shallow wait-for-done filter
15:39:56.617346 pkt-line.c:80           packet:        clone< server-option
15:39:56.617350 pkt-line.c:80           packet:        clone< object-format=sha1
15:39:56.617354 pkt-line.c:80           packet:        clone< 0000
15:39:56.617370 pkt-line.c:80           packet:        clone> command=ls-refs
15:39:56.617394 pkt-line.c:80           packet:        clone> agent=git/2.39.3.(Apple.Git-146)
15:39:56.617402 pkt-line.c:80           packet:        clone> object-format=sha1
15:39:56.617406 pkt-line.c:80           packet:        clone> 0001
15:39:56.617410 pkt-line.c:80           packet:        clone> peel
15:39:56.617414 pkt-line.c:80           packet:        clone> symrefs
15:39:56.617419 pkt-line.c:80           packet:        clone> unborn
15:39:56.617427 pkt-line.c:80           packet:        clone> ref-prefix HEAD
15:39:56.617432 pkt-line.c:80           packet:        clone> ref-prefix refs/heads/
15:39:56.617436 pkt-line.c:80           packet:        clone> ref-prefix refs/tags/
15:39:56.617457 pkt-line.c:80           packet:        clone> 0000
15:39:56.785581 pkt-line.c:80           packet:        clone< be8cb94880285b13b7bbcf359e26b6655832f0ab HEAD symref-target:refs/heads/master
15:39:56.785656 pkt-line.c:80           packet:        clone< be8cb94880285b13b7bbcf359e26b6655832f0ab ref
...
```

**Comment by syntonyze** [comment]
https://eclipse.gerrithub.io/c/eclipse-jgit/jgit/+/1205032

**Comment by msohn** [comment]
merged as 2b229df06c94fa351ce00227a107835e3b21f5ce
